pkg/container-collection/container-collection.go

```diff
@@ -339,7 +370,12 @@ func (cc *ContainerCollection) EnrichByMntNs(event *eventtypes.CommonData, mount
 func (cc *ContainerCollection) EnrichByNetNs(event *eventtypes.CommonData, netnsid uint64) {
 	event.Node = cc.nodeName
```

Comment:
I'm a bit worried about performance here, since the code is pretty hot: adding a second map to range over for every incoming event (unlike in EnrichByMntNs, where it's probably used rarely when the container hasn't been found) including handling locks etc. seems pretty expensive.

What about adding a `deleted` flag to the container struct and filter using that instead of using a different map?

However, in the long term, we should probably add two new maps for directly looking up by MntNs (e.g. `map[uint64]*Container`) & NetNs (something like `map[uint64]struct{Container *Container, useCount int}`, where useCount is used to do skip the ContainerName when enriching) which are updated when containers are added/deleted, as ranging over potentially a couple of hundred containers in a production environment could really hit performance.